File: java/CoffeeMachine.java
import java.util.Scanner;

public class CoffeeMachine {
    
    public static final int WATER_PER_CUP_ESPRESSO = 250;
    public static final int MILK_PER_CUP_ESPRESSO = 0;
    public static final int COFFEE_BEANS_PER_CUP_ESPRESSO = 16;
    public static final int MONEY_PER_CUP_ESPRESSO = 4;
    
    public static final int WATER_PER_CUP_LATTE = 350;
    public static final int MILK_PER_CUP_LATTE = 75;
    public static final int COFFEE_BEANS_PER_CUP_LATTE = 20;
    public static final int MONEY_PER_CUP_LATTE = 7;
    
    public static final int WATER_PER_CUP_CAPPUCCINO = 200;
    public static final int MILK_PER_CUP_CAPPUCCINO = 100;
    public static final int COFFEE_BEANS_PER_CUP_CAPPUCCINO = 12;
    public static final int MONEY_PER_CUP_CAPPUCCINO = 6;
    
    public static int water = 400;
    public static int milk = 540;		
    public static int coffeeBeans = 120;
    public static int cups = 9;
    public static int money = 550;
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        
        String word;
        
        while(true) {
        	System.out.println("Write action (buy, fill, take, remaining, exit):");
        	word = scanner.next();
	        switch (word) {
	            case "fill":
	            	fill(scanner);
	            	continue;
	            case "take":
	            	take();
	            	continue;
	            case "buy":
	            	buy(scanner);
	            	continue;
	            case "remaining":
	            	machineState();
	            	continue;
	            default:
	            	return;
	        }
        }
    }
    
    public static void machineState() {
    	System.out.printf("The coffee machine has:\n%d ml of water\n%d ml of milk\n%d g of coffee beans\n%d "
    			+ "disposable cups \n$%d of money\n\n", water, milk, coffeeBeans, cups, money);
    }
    
    public static void fill(Scanner scanner) {
    	System.out.println("Write how many ml of water you want to add:");
        water += scanner.nextInt();
        System.out.println("Write how many ml of milk you want to add:");
        milk += scanner.nextInt();
        System.out.println("Write how many grams of coffee beans you want to add:");
        coffeeBeans += scanner.nextInt();
        System.out.println("Write how many disposable cups of coffee you want to add:");
        cups += scanner.nextInt();
    }
    
    public static void take() {
    	System.out.printf("I gave you $%d\n\n", money);
    	money = 0;
    }
    
    public static void buy(Scanner scanner) {
    	System.out.println("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino, back - to main menu: ");
    	String x = scanner.next();
    	
    	switch (x) {
    	    case "1":
    	    	if (water - WATER_PER_CUP_ESPRESSO < 0) {
    	    		System.out.println("Sorry, not enough water!\n");
    	    		return;
    	    	} else if (coffeeBeans - COFFEE_BEANS_PER_CUP_ESPRESSO < 0) {
    	    		System.out.println("Sorry, not enough coffee beans!\n");
    	    		return;
    	    	} else if (cups - 1 < 0) {
    	    		System.out.println("Sorry, not enough cups!\n");
    	    		return;
    	    	} else {
    	    		System.out.println("I have enough resources, making you a coffee!");
	    	    	water -= WATER_PER_CUP_ESPRESSO;
	    	    	coffeeBeans -= COFFEE_BEANS_PER_CUP_ESPRESSO;
	    	    	cups--;
	    	    	money += MONEY_PER_CUP_ESPRESSO;
	    	    	break;
    	    	}	
    	    case "2":
    	    	if (water - WATER_PER_CUP_LATTE < 0) {
    	    		System.out.println("Sorry, not enough water!\n");
    	    		return;
    	    	} else if (milk - MILK_PER_CUP_LATTE < 0) {
    	    		System.out.println("Sorry, not enough milk!\n");
    	    		return;
    	    	} else if (coffeeBeans - COFFEE_BEANS_PER_CUP_LATTE < 0) {
    	    		System.out.println("Sorry, not enough coffee beans!\n");
    	    		return;
    	    	} else if (cups - 1 < 0) {
    	    		System.out.println("Sorry, not enough cups!\n");
    	    		return;
    	    	} else {
    	    		System.out.println("I have enough resources, making you a coffee!");
	    	    	water -= WATER_PER_CUP_LATTE;
	    	    	milk -= MILK_PER_CUP_LATTE;
	    	    	coffeeBeans -= COFFEE_BEANS_PER_CUP_LATTE;
	    	    	cups--;
	    	    	money += MONEY_PER_CUP_LATTE;
	    	    	break;
    	    	}	
    	    case "3":
    	    	if (water - WATER_PER_CUP_CAPPUCCINO < 0) {
    	    		System.out.println("Sorry, not enough water!\n");
    	    		return;
    	    	} else if (milk - MILK_PER_CUP_CAPPUCCINO < 0) {
    	    		System.out.println("Sorry, not enough milk!\n");
    	    		return;
    	    	} else if (coffeeBeans - COFFEE_BEANS_PER_CUP_CAPPUCCINO < 0) {
    	    		System.out.println("Sorry, not enough coffee beans!\n");
    	    		return;
    	    	} else if (cups - 1 < 0) {
    	    		System.out.println("Sorry, not enough cups!\n");
    	    		return;
    	    	} else {
    	    		System.out.println("I have enough resources, making you a coffee!");
	    	    	water -= WATER_PER_CUP_CAPPUCCINO;
	    	    	milk -= MILK_PER_CUP_CAPPUCCINO;
	    	    	coffeeBeans -= COFFEE_BEANS_PER_CUP_CAPPUCCINO;
	    	    	cups--;
	    	    	money += MONEY_PER_CUP_CAPPUCCINO;
	    	    	break;
    	    	}	
    	    default:
    	    	System.out.println();
    	    	return;
    	}
    	System.out.println();
    }
}

File: python/coffee_machine.py
water = 400
milk = 540
coffee_beans = 120
cups = 9
money = 550


def information():
    print("The coffee machine has:")
    print(water, "of water")
    print(milk, "of milk")
    print(coffee_beans, "of coffee beans")
    print(cups, "of cups")
    print(money, "of money")


def take():
    global money
    print("I gave you $" + str(money))
    money = 0


def fill():
    global water, milk, coffee_beans, cups
    print("Write how many ml of water you want to add:")
    water += int(input())
    print("Write how many ml of milk you want to add:")
    milk += int(input())
    print("Write how many grams of coffee beans you want to add:")
    coffee_beans += int(input())
    print("Write how many disposable coffee cups you want to add:")
    cups += int(input())


def buy():
    global water, milk, coffee_beans, cups, money
    print("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino, back - to main menu:")
    choice = input()
    if cups < 1 and choice != "back":
        print("Sorry, not enough disposable cups!")
        return
    if choice == "1":
        if water < 250:
            print("Sorry, not enough water!")
            return
        elif coffee_beans < 16:
            print("Sorry, not enough coffee beans!")
            return
        else:
            water -= 250
            coffee_beans -= 16
            money += 4
    elif choice == "2":
        if water < 350:
            print("Sorry, not enough water!")
            return
        elif milk < 75:
            print("Sorry, not enough milk!")
            return
        elif coffee_beans < 20:
            print("Sorry, not enough coffee beans!")
            return
        else:
            water -= 350
            milk -= 75
            coffee_beans -= 20
            money += 7
    elif choice == "3":
        if water < 200:
            print("Sorry, not enough water!")
            return
        elif milk < 100:
            print("Sorry, not enough milk!")
            return
        elif coffee_beans < 12:
            print("Sorry, not enough coffee beans!")
            return
        else:
            water -= 200
            milk -= 100
            coffee_beans -= 12
            money += 6
    else:
        cups += 1
    cups -= 1


while True:
    print("Write action (buy, fill, take, remaining, exit):")
    word = input()
    if word == "buy":
        buy()
    elif word == "fill":
        fill()
    elif word == "take":
        take()
    elif word == "remaining":
        information()
    else:
        break
